### DISCUSSION TITLE
Highlighting `support.class` scope (example on PlantUML language)

### DISCUSSION THREAD

**Discussion by The-Lum** [github-linguist/linguist#5530] [Q&A] [answered]
Hello,

We have some questions about language highlighting.

---
Discussion inspired from:
- qjebbs/vscode-plantuml#428

For example, if we take the PlantUML language with the corresponding syntax definition: 
- https://github.com/qjebbs/vscode-plantuml
- https://github.com/qjebbs/vscode-plantuml/blob/a151047641c224772d6553d98ccfe1b60483c60a/syntaxes/plantuml.yaml-tmLanguage#L202
---

- Why the `support.class.preprocessings.source.wsd` keywords are not highlighting on Github?
_while very well on VS Code..._

See keyword `!include` on this simple example:
```puml
@startuml
!include filename.puml
@enduml
```

- How to differentiate explicitly the **_textmate scope_**  `support.class` on Github?
- Where is the issue?

Regards.

**Comment by lildude** [comment, marked as answer]
> * Why the `support.class.preprocessings.source.wsd` keywords are not highlighting on Github?

Well they technically are (they're not black/white depending on dark mode) but only because GitHub only implements limited support for `support` keywords with `support.class` _not_ being one of them.

@Alhadis  maintains a great list showing the supported keywords and their colours in https://github.com/Alhadis/language-etc/blob/master/samples/lists/scope-previews.nanorc

Also see his comment at https://github.com/github/linguist/pull/4568#issuecomment-513739638 for more details on how GitHub behaves.

**Reply by The-Lum** [reply]
Thanks a lot.

Your other answer on https://github.com/github/linguist/discussions/5204#discussioncomment-352539 is also helpful.

Then, I will look on the **theme** side (_to ask if a explicit differentiation is plan_) or on the initial language **grammar** (_in order to change the **scope** of the preprocessing keywords_).

**Reply by Alhadis** [reply]
I see this part of my comment was accurate:

> Don't get too attached to the colours you see. […] Expect the exact colours to shift over time.

… because today I noticed that GitHub's `variable` colour has changed from `#E36209` to `#953800`. 😉
